### PR TITLE
Add prettifier type for the output type

### LINKED ISSUE #23493: SDK: prettify rest command output type in IDE
### Describe the Improvement

Output type of the `readItem` or similar commands is hardly readable in the IDE. Take a look at this example:

```typescript
import { type CompleteSchema, type DirectusUser, createDirectus, rest, readItem } from "@directus/sdk";

type TestEntity = {
    id: string;
    user: DirectusUser<Schema>;
    name: string;
    nickname: string | undefined;
};

type Schema = CompleteSchema<{
    testEntity: TestEntity[] ;
}>;

const client = createDirectus<Schema>("").with(rest());

const data = await client.request(readItem("testEntity", "1", {
    fields: ["id", "nickname", { user: ["id"] }]
}));
```

If I hover over the `data` variable to see the type, VSCode will show me this:

image

There is a way to improve DX by "prettifying" complex types by leveraging a saving of "computed" type via TS mapping types (see this article for details: https://www.totaltypescript.com/concepts/the-prettify-helper).

I've already implemented this in my fork of Directus (see [commit](https://github.com/directus/directus/pull/23494/commits/c521ec4a9239ac30bf0ee9a55681b61731e5f21e)) and here what I got for the same code snippet:
image

I think this change might improve DX without any drawbacks and also simplify usage of type aliases to stay in sync with actual data shape that we get from the API when using `fields` in the query of a command.

### PR DESCRIPTION
**Pull request by fapspirit**
## Scope

What's changed:

- Added prettifier for the rest command output types so that IDE will show the only actual computed type instead of bunch of generics that is used to compute the type

## Potential Risks / Drawbacks

- N/A

## Review Notes / Questions

- Would like to hint on any edge cases that I'm not aware of

---

Fixes #23493

### CODE REVIEW

File: sdk/src/types/utils.ts [hunk 1]
```diff
@@ -52,3 +52,7 @@ export type NestedPartial<Item extends object> = {
  * Resolve type to its final object
  */
 export type Identity<U> = U extends infer A ? A : U;
+
+export type Prettify<T> = {
```
Comment (paescuj) [inline sdk/src/types/utils.ts:56]: Let's add a JSDoc description:

```suggestion
/**
 * Reduces a complex object type to make it readable in IDEs.
 */
export type Prettify<T> = {
```

File: sdk/src/types/output.ts [hunk 2]
```diff
@@ -17,29 +17,31 @@ export type ApplyQueryFields<
...
+										RelatedCollection extends any[]
+											? HasManyToAnyRelation<RelatedCollection> extends never
+												? ApplyNestedQueryFields<Schema, RelatedCollection, RelationalFields[Field]>[] | null // many-to-many or one-to-many
+												: ApplyManyToAnyFields<Schema, RelatedCollection, RelationalFields[Field]>[] // many-to-any'
+											: ApplyNestedQueryFields<Schema, RelatedCollection, RelationalFields[Field]> // many-to-one
+								  >
+								: never
+							: never;
+				  }
+		>
```
[the remark this answers is not in the document]
Comment (fapspirit) [reply sdk/src/types/output.ts]: Makes sense, fixed

File: sdk/src/types/output.ts [hunk 3]
```diff
@@ -53,21 +55,23 @@ export type ApplyManyToAnyFields<
...
+							item: {
+								[Scope in keyof ItemFields]: Scope extends keyof Schema
+									? ApplyNestedQueryFields<Schema, Schema[Scope], ItemFields[Scope]>
+									: never;
+							}[keyof ItemFields];
+					  }
+					: never
+			  : ApplyQueryFields<Schema, Junction, Readonly<UnpackList<FieldsList>>> // no items query
+		: never
+>;
```
[the remark this answers is not in the document]
Comment (fapspirit) [reply sdk/src/types/output.ts]: Thank you for the hint! I initially thought the Prettifier type might struggle with recursive types, but I tested it around and seems it works without prettifying in this specific type.

**Review by paescuj — APPROVED** [review]
Thanks @fapspirit ❤️

**Pull request merged**